### PR TITLE
Replace vulnerable pom-parser with fast-xml-parser in tests, and other updates

### PR DESCRIPTION
**Pull request by fbricon**
Signed-off-by: Fred Bricon <fbricon@gmail.com>

### CODE REVIEW

File: src/test/vscodeTest/suite/index.ts [hunk 1]
```diff
@@ -11,27 +11,14 @@ export function run(): Promise<void> {
...
-          } else {
-            c();
-          }
-        });
-      } catch (err) {
-        e(err);
-      }
-    });
+  // Run the mocha test
+  mocha.run(failures => {
```
Comment (datho7561) [inline src/test/vscodeTest/suite/index.ts]: `mocha.run` will return before the callback is run, which means the program will exit without reporting any passes or failures. To get around this, you need to wrap it in a Promise and await that promise.

See https://github.com/redhat-developer/vscode-quarkus/pull/648/files#diff-5116d4bcabdc6045aa26cc1afdd56ab3155276c7d69ff9b3be07b9fb157da247R19

**Review by datho7561 — CHANGES_REQUESTED** [review]

File: src/test/vscodeTest/suite/index.ts (hunk 1 above)
Comment (fbricon) [reply src/test/vscodeTest/suite/index.ts]: right. I still suck at Promises ;-)

File: package.json [hunk 2]
```diff
@@ -34,21 +34,14 @@
...
     "workspaceContains:**/src/main/resources/application.yaml",
     "workspaceContains:**/src/main/resources/application.yml",
     "workspaceContains:**/src/main/resources/templates",
     "onLanguage:microprofile-properties",
-    "onLanguage:quarkus-properties",
-    "onLanguage:java",
-    "onLanguage:qute-html",
-    "onLanguage:qute-json",
-    "onLanguage:qute-yaml",
-    "onLanguage:qute-txt"
```
Comment (fbricon) [inline package.json:51]: onCommand and onLanguage events are automatically registered, no need to declare them explicitly here

File: src/test/vscodeTest/suite/extension.test.ts [hunk 3]
```diff
@@ -10,17 +10,4 @@ describe('VS Code extension tests', () => {
...
-    const activationCommands = packageJSON.activationEvents.filter((s) => s.startsWith('onCommand:'))
-        .map((s) => s.substring('onCommand:'.length));
-    const QUARKUS_ACTIVATION_COMMANDS: string[] = [
-      // from package.json
-      VSCodeCommands.CREATE_PROJECT,
-      VSCodeCommands.QUARKUS_WELCOME,
-    ];
-
-    expect(activationCommands).to.have.members(QUARKUS_ACTIVATION_COMMANDS);
-  });
```
Comment (fbricon) [inline src/test/vscodeTest/suite/extension.test.ts:25]: onCommand events are automatically registered, no need to test them here

**Comment by fbricon** [comment]
tests finally pass

File: package.json [hunk 4]
```diff
@@ -442,14 +435,15 @@
     "@types/chai-fs": "^2.0.2",
     "@types/ejs": "^3.0.5",
     "@types/fs-extra": "^7.0.0",
+    "@types/glob": "^8.1.0",
```
Comment (datho7561) [inline package.json]: glob 9+ comes with the type declarations built into the main package, so we don't need this package.

**Review by datho7561 — CHANGES_REQUESTED** [review]
I think this is ready to merge after this comment is addressed

File: package.json (hunk 4 above)
Comment (fbricon) [reply package.json]: For some reason it didn't find the types until I added them

Comment (datho7561) [reply package.json]: It seems to work for me. What was reporting the error, VS Code or compiling from command line?

Comment (fbricon) [reply package.json]: VS Code. Will try again tomorrow

Comment (fbricon) [reply package.json]: it works now

**Review by datho7561 — APPROVED** [review]
Looks good. Thanks, Fred!

**Pull request merged**